tests/actions/test_core_actions.py

```diff
@@ -533,39 +533,51 @@ def test_SlackNotificationAction_run(self, checkpoint_result: CheckpointResult):
         with mock.patch.object(Session, "post") as mock_post:
             output = action.run(checkpoint_result=checkpoint_result)
 
-        assert mock_post.call_count == 5  # Sent in batches
+        assert mock_post.call_count == 1
         mock_post.assert_called_with(
```

Comment:
would changing this to `aasert_called_once_with` have caught the bug before? Either way, we should probably change it now